### DISCUSSION TITLE
Does the {gameType} token work for anyone else?

### DISCUSSION THREAD

**Discussion by samwelnella** [emmercm/igir#688] [Q&A] [answered]
Hello, I'm finding that when I use the {gameType} all games end up in the "Retail" folder including games that have "(Proto)", "(Pirate)" or "(Aftermarket)" in the ROM name. Does anyone else have this issue?

**Comment by emmercm** [comment]
Hi @samwelnella, this definitely isn't expected. Could you provide some additional information about:

- What DATs you're using, if any
- What some example input files are
- What other options you're using, especially any filtering ones
- What `igir` version you're using, if it's the most recent version or not

**Reply by samwelnella** [reply]
Hi @emmercm!

To answer your questions in order

- I am using No-Intro Parent Child XML DATs from dat-o-matic
- Some examples are: "Moto-X (World) (Aftermarket) (Unl).zip", "X-Men vs. Street Fighter (USA) (Pirate).zip" and "Apocalypse II (Europe) (Proto 2) (1997-10-28).zip"
- I am using the --single, --prefer-language and --prefer-region options as in the 1g1r example
- I am using igir version 2.0

Thank you for your help!

**Reply by samwelnella** [reply]
I should add I was seeing the same behaviour with 1.9.3

**Reply by emmercm** [reply]
I just tried myself and it worked as expected:

```shell
$ npx --yes igir@latest copy -d "No-Intro*.zip" -i "../Nintendo - Super Nintendo Entertainment System/**" -o "./{gameType}/" --single --prefer-language EN --prefer-region USA,WORLD,EUR,JPN
 ______   ______   ______  _______  
|      \ /      \ |      \|       \ 
 \$$$$$$|  $$$$$$\ \$$$$$$| $$$$$$$\
  | $$  | $$|    \  | $$  | $$    $$   ROM collection manager
  | $$  | $$|    \  | $$  | $$    $$   https://igir.io/
  | $$  | $$ \$$$$  | $$  | $$$$$$$\
 _| $$_ | $$__| $$ _| $$_ | $$  | $$   v2.0.0
|   $$ \ \$$    $$|   $$ \| $$  | $$
 \$$$$$$  \$$$$$$  \$$$$$$ \$$   \$$


✓ Scanning for DATs ·········· | 286 unique DATs found
✓ Scanning for ROMs ·········· | 3,990 files found
✓ Nintendo - Super Nintendo En | 1,933/1,990 games, 1/1 BIOSes, 1,782/1,798 retail releases written


$ du -h
 16M	./Beta
1.9M	./Demo
1.0M	./Unlicensed
1.5G	./Retail
9.2M	./Prototype
 21M	./Pirated
756K	./Sample
 36K	./BIOS
 14M	./Aftermarket
1.6G	.
```

could you paste your full command?

**Reply by samwelnella** [reply]
Sure my full command is

`igir copy extract --dat /srv/mergerfs/PoolFS/Games/DATs/PC\ XML/No-Intro/Nintendo\ -\ Game\ Boy\ Color\ \(Parent-Clone\)\ \(20230907-191751\).dat --input /srv/mergerfs/PoolFS/Games/No-Intro/Nintendo\ -\ Game\ Boy\ Color --output /srv/mergerfs/PoolFS/Games/Organized/gbc/{gameType}/{datReleaseRegion}/ --single --prefer-language EN --prefer-region USA,WORLD,EUR,JPN`

I wonder if there's something wrong with my DAT files?

**Reply by emmercm** [reply]
I would be surprised if it was the DAT, that looks like a fresh download, I'm assuming it's from https://datomatic.no-intro.org/index.php?page=download&op=xml&s=47.

I wonder if your shell is doing some globbing with your output path not being quoted, I'd try quoting that.

Could you provide some checksums of the extracted output files (the path isn't quoted in purpose):

```shell
crc32 /srv/mergerfs/PoolFS/Games/Organized/gbc/*/*/*({Proto,Pirate,Aftermarket})* | head -10
```

**Reply by samwelnella** [reply]
I am actually using the daily P/C XML DATs from the daily tab here https://datomatic.no-intro.org/index.php?page=download&op=daily&s=47

I tried quoting the paths but I still had the same result.

Here's the result of the crc32 command:

```
8f3902f2        /srv/mergerfs/PoolFS/Games/Organized/gbc/BIOS/TAI/[BIOS] Genius GB Backup Station (Taiwan) (Unl).gbc
4a5f1e2b        /srv/mergerfs/PoolFS/Games/Organized/gbc/BIOS/TAI/[BIOS] Touch Boy System ROM (Taiwan) (Unl).gbc
59579caa        /srv/mergerfs/PoolFS/Games/Organized/gbc/BIOS/USA/[BIOS] Challenger GB (USA) (Unl).gbc
41884e46        /srv/mergerfs/PoolFS/Games/Organized/gbc/BIOS/USA/[BIOS] Nintendo Game Boy Color Boot ROM (World) (Rev 1).gbc
21524051        /srv/mergerfs/PoolFS/Games/Organized/gbc/Retail/AUS/31-in-1 Mighty Mix (Australia) (31B-001, Sachen) (Unl).gbc
a4ad3678        /srv/mergerfs/PoolFS/Games/Organized/gbc/Retail/CHN/Shuihu Zhuan Zhi Qunmo Fengyun Lu (China) (Unl).gbc
188b06f8        /srv/mergerfs/PoolFS/Games/Organized/gbc/Retail/CHN/Terrifying 911 (China) (En) (Unl).gbc
5046c427        /srv/mergerfs/PoolFS/Games/Organized/gbc/Retail/EUR/Absolute X (Europe) (Proto).gbc
c288e400        /srv/mergerfs/PoolFS/Games/Organized/gbc/Retail/EUR/Action Replay Xtreme - Special Edition for Pokemon Crystal (Europe) (Unl).gbc
d0d3dfed        /srv/mergerfs/PoolFS/Games/Organized/gbc/Retail/EUR/Adventures of the Smurfs, The (Europe) (En,Fr,De,Es,It,Nl).gbc
```

**Reply by emmercm** [reply, marked as answer]
I found the issue. I was making the assumption that if a game had any releases it would be "verified" and verified is considered retail. `Absolute X` has a release, so it's erroneously considered retail:

```xml
<game name="Absolute X (Europe) (Proto)">
  <description>Absolute X (Europe) (Proto)</description>
  <release name="Absolute X (Europe) (Proto)" region="EUR"/>
  <rom name="Absolute X (Europe) (Proto).gbc" size="1048576" crc="5046c427" md5="cffbcb83ed97a4ff88f71ee383bd0eba" sha1="34fa9bc6b1fec75751f1f426528f18c833128ff7"/>
</game>
```

I'll convert this into a bug report and put in the PR.
